applications/GeoMechanicsApplication/custom_utilities/element_utilities.hpp

```diff
@@ -697,6 +594,27 @@ typedef std::size_t IndexType;
         return nodal_hydraulic_heads;
     }
 
+private:
+    template <typename VectorType1, typename VectorType2>
+    static void AddVectorAtPosition(const VectorType1& rSourceVector, VectorType2& rDestinationVector, std::size_t Offset)
+    {
+        auto pos = std::begin(rDestinationVector) + Offset;
+        std::transform(std::begin(rSourceVector), std::end(rSourceVector), pos, pos, std::plus<double>{});
+    }
+
+    template <typename MatrixType1, typename MatrixType2>
+    static void AddMatrixAtPosition(const MatrixType1& rSourceMatrix,
+                                    MatrixType2&       rDestinationMatrix,
+                                    std::size_t        RowOffset,
+                                    std::size_t        ColumnOffset)
+    {
+        for (auto i = std::size_t{0}; i < rSourceMatrix.size1(); ++i) {
+            for (auto j = std::size_t{0}; j < rSourceMatrix.size2(); ++j) {
+                rDestinationMatrix(i + RowOffset, j + ColumnOffset) += rSourceMatrix(i, j);
+            }
+        }
```

Comment:
Should we do a check somewhere such that we're sure that we don't go out of bounds of the destination matrix? (e.g. source is 3x3, destination 4x4 and offsets are 2)? I'm not sure what happens when using transform and such a thing happens (in the cases of the vector addition)

Reply:
I believe that such a check is not really needed here. Also the STL algorithms in general don't carry out such checks. They assume it is the caller's responsibility to ensure that the operation won't attempt to access data that is out of bounds. If the programmer forgets about that, you enter the domain of undefined behavior, which may or may not result in a segfault. I would keep it the way it is now.